FirebaseSessions/README.md

```diff
@@ -42,3 +50,11 @@ brew install protobuf
  1. Follow the directions in `sessions.proto` for updating it
  1. Run the following to regenerate the nanopb source files: `./FirebaseSessions/ProtoSupport/generate_protos.sh`
  1. Update the SDK to use the new proto fields
+
+
+### Logging
+The Sessions SDK uses the following strategy when determining log level:
+ - **Info** should be used rarely. Because the Sessions SDK is a dependency of other products, customers will not expect regular logs from the SDK. Therefore, info events are not recommended except under circumstances where the code path is blocked by another debug parameter (eg. `-FIRSessionsDebugEvents` will log under info because we don't want to require it be paired with `-FIRDebugEnabled`)
+ - **Debug** Is recommended to be used generously in the Sessions SDK for the purposes of debugging customer issues.
+ - **Warning** Is used when the Sessions SDK runs into a recoverable issue that still results in events being sent. For example, a problem converting between values that results in an incorrect value being reported.
+ - **Error** Is used when the Sessions SDK runs into an unrecoverable issue that prevents functionality from working. If we would want customers to reach out to us when a issue happens, then error logs should be used to convey the issue.
```

Comment:
How do these levels maps to issues that are "Fixable" by the developer themselves versus the ones that the developers would be aware, but they cannot do anything about it?

For eg: Error state IMO is something that the developer cannot fix by themselves and unaware of the impact of the message. Warning is something that the developer cannot do anything about, but has a sense of the impact. Debug is something to give more details for the developers to share with us.

It its current definition, the term "unrecoverable" is something from the SDK perspective, but does not provide any signal/actionable thing from the developers perspective. Is that how you imagine this to be?

Reply:
So I've been thinking about this more, and I think we need to use language to convey fixability instead of log level.

Eg. Say we decided `warning` was for fixable problems, but there was an issue that would cause the SDK to not upload events. In that situation I think I would want the log to be an error so that it's more visible to the customer.

So maybe our recommendation should be: if it's fixable, put the resolution steps in the log itself.

Reply:
That sounds good. I like the verbiage nature of these errors/concerns that can enable the developers to fix it. In that case, does an `Error` level mean that it is something the developer can fix it? Or should such issues be tagged at `Warning` level.

Reply:
As long as that is aligned - the changes look good to me.

Reply:
Yeah what I meant above is to make it based on the message instead of log level.